test/k8sT/hubble.go

```diff
@@ -187,6 +190,13 @@ var _ = Describe("K8sHubbleTest", func() {
 				"hubble observe --last 1 --json --type trace --from-pod %s/%s --to-namespace %s --to-label %s --to-port %d",
 				namespaceForTest, appPods[helpers.App2], namespaceForTest, app1Labels, app1Port), `"Type":"L3_L4"`)
 			Expect(err).To(BeNil(), "hubble observe query timed out")
+
+			// Basic check for L4 Prometheus metrics.
+			_, nodeIP := kubectl.GetNodeInfo(helpers.K8s1)
+			Expect(nodeIP).NotTo(BeEmpty(), "failed to get node IP for %s", helpers.K8s1)
+			err = hubbleExecUntilMatch(hubbleNamespace, hubblePodK8s1, fmt.Sprintf("curl %s:%d/metrics", nodeIP, prometheusPort),
```

Comment:
The overall logic of the invocation itself looks good to me.

`hubbleExecUntilMatch` however got removed in #11213. I would recommend using a construct along these lines (untested code)
```go
metricsUrl := fmt.Sprintf("%s/metrics", net.JoinHostPort(nodeIP, prometheusPort)
// this will block until curl finished
res := kubectl.ExecPodCmd(ns, pod, helpers.CurlFail(metricsUrl))
res.ExpectSuccess("%s/%s cannot curl metrics %q", ns, pod, app1ClusterIP)
res.ExpectContains(`hubble_flows_processed_total{subtype="to-endpoint",type="Trace",verdict="FORWARDED"}`)
```